discovery/kubernetes_api_discovery.go

```diff
@@ -222,13 +222,18 @@ func (k *K8sAPIDiscoverer) getServices() ([]byte, error) {
 		return data, err
 	}
 
-	k.lock.Lock()
+	discoveredSvcs := make(map[string]*K8sService)
+
 	for _, s := range svcs.Items {
 		// Avoid Go loop var pointer re-use
 		svc := s
-		k.discoveredSvcs[svc.ServiceName()] = &svc
+		discoveredSvcs[svc.ServiceName()] = &svc
 	}
+
+	k.lock.Lock()
+	k.discoveredSvcs = discoveredSvcs
```

Comment:
Woot! Major reduction in lock time. Brings CPU down a lot by removing spin waits.